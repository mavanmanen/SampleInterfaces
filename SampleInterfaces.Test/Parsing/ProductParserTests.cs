using ApprovalTests.Reporters;
using Newtonsoft.Json;
using SampleInterfaces.Models;
using SampleInterfaces.Parsing;
using SampleInterfaces.Test.Helpers;

namespace SampleInterfaces.Test.Parsing;

[UseReporter(typeof(DiffReporter))]
public class ProductParserTests
{
    [Fact]
    public void Parse_HappyFlow_Succeeds()
    {
        // Arrange
        var sut = new ProductParser();
        var input = JsonConvert.SerializeObject(new Product[]
        {
            new(1, "Product 1", 12.30m),
            new(2, "Product 2", 52.23m),
            new(3, "Product 3", 42.12m)
        });
        
        // Act
        var result = sut.Parse(input);
        
        // Assert
        ApprovalsHelper.VerifyObject(result);
    }

    [Theory]
    [InlineData("")]
    [InlineData(" ")]
    [InlineData(null)]
    public void Parse_InputValueInvalid_ExceptionThrown(string input)
    {
        // Arrange
        var sut = new ProductParser();
        
        // Act
        IEnumerable<Product> Action() => sut.Parse(input);
        
        // Assert
        Assert.Throws<ArgumentNullException>(nameof(input), Action);
    }
}