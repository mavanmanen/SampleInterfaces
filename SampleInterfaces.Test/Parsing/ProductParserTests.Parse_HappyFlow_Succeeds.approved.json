[
    {
        "Id": 1,
        "Name": "Product 1",
        "Price": 12.30
    },
    {
        "Id": 2,
        "Name": "Product 2",
        "Price": 52.23
    },
    {
        "Id": 3,
        "Name": "Product 3",
        "Price": 42.12
    }
]